sdk/core/azure-core/src/main/java/com/azure/core/http/ProxyOptions.java

```diff
@@ -68,6 +115,125 @@ public String getPassword() {
+     */
+    public static ProxyOptions loadFromEnvironment(Configuration configuration) {
+        if (configuration == Configuration.NONE) {
+            throw new IllegalArgumentException("'configuration' cannot be 'Configuration.NONE'.");
+        }
+
+        Configuration proxyConfiguration = (configuration == null)
+            ? Configuration.getGlobalConfiguration()
+            : configuration;
+
+        for (Function<Configuration, ProxyOptions> loader : ENVIRONMENT_LOAD_ORDER) {
+            ProxyOptions proxyOptions = loader.apply(proxyConfiguration);
+            if (proxyOptions != null) {
+                return proxyOptions;
+            }
+        }
+
+        return null;
+    }
+
+    private static ProxyOptions attemptToLoadAzureProxy(Configuration configuration, String proxyProperty) {
+        String proxyConfiguration = configuration.get(proxyProperty);
+
+        // No proxy configuration setup.
+        if (CoreUtils.isNullOrEmpty(proxyConfiguration)) {
+            return null;
+        }
+
+        try {
+            URL proxyUrl = new URL(proxyConfiguration);
+            int port = (proxyUrl.getPort() == -1) ? proxyUrl.getDefaultPort() : proxyUrl.getPort();
+            ProxyOptions proxyOptions = new ProxyOptions(Type.HTTP, new InetSocketAddress(proxyUrl.getHost(), port))
+                .setNonProxyHosts(configuration.get(Configuration.PROPERTY_NO_PROXY));
+
+            String userInfo = proxyUrl.getUserInfo();
+            if (userInfo != null) {
+                String[] usernamePassword = userInfo.split(":", 2);
+                if (usernamePassword.length == 2) {
+                    try {
+                        proxyOptions.setCredentials(
+                            URLDecoder.decode(usernamePassword[0], StandardCharsets.UTF_8.toString()),
+                            URLDecoder.decode(usernamePassword[1], StandardCharsets.UTF_8.toString())
+                        );
+                    } catch (UnsupportedEncodingException e) {
+                        return null;
+                    }
+                }
+            }
+
+            return proxyOptions;
+        } catch (MalformedURLException ex) {
+            return null;
+        }
+    }
+
+    private static ProxyOptions attemptToLoadJavaProxy(Configuration configuration, String type) {
+        // Not allowed to use Java proxies
+        if (!Boolean.parseBoolean(configuration.get(JAVA_PROXY_PREREQUISITE))) {
+            return null;
+        }
+
+        String host = configuration.get(type + "." + JAVA_PROXY_HOST);
+
+        // No proxy configuration setup.
+        if (CoreUtils.isNullOrEmpty(host)) {
+            return null;
+        }
+
+        int port;
+        try {
+            port = Integer.parseInt(configuration.get(type + "." + JAVA_PROXY_PORT));
+        } catch (NumberFormatException ex) {
+            port = HTTPS.equals(type) ? 443 : 80;
```

Comment:
Consider making these named constants `DEFAULT_HTTPS_PORT` and `DEFAULT_HTTP_PORT`.

Reply:
Made these constant.